sorc/ncep_post.fd/MISCLN.f

```diff
@@ -2080,17 +2080,40 @@ SUBROUTINE MISCLN
            DPBND = 0.
            CALL CALCAPE(ITYPE,DPBND,P1D,T1D,Q1D,LB2,EGRID1,   &
                         EGRID2,EGRID3,EGRID4,EGRID5) 
+
 !
-           IF (IGET(566)>0) THEN
-! dong add missing value for cape
+           IF(IGET(566)>0 .or. NEED_IFI) THEN
               GRID1=spval
 !$omp parallel do private(i,j)
               DO J=JSTA,JEND
                 DO I=ISTA,IEND
                   IF(T1D(I,J) < spval) GRID1(I,J) = EGRID1(I,J)
                 ENDDO
               ENDDO
+              CALL BOUND(GRID1,D00,H99999)
+!$omp parallel do private(i,j)
+              DO J=JSTA,JEND
+                DO I=ISTA,IEND
+                  CAPE(I,J) = GRID1(I,J)
+                ENDDO
+              ENDDO
+           ENDIF
+
+           IF(IGET(567)>0 .or. NEED_IFI) THEN
```

Comment:
@SamuelTrahanNOAA It seems to me the line 2102 to 2114 overlap the changes of the block starting from line 2132. Can these lines be removed?

Reply:
I'm not sure what you mean. The code may look similar at a glance, but they do different things.

Reply:
@SamuelTrahanNOAA You assign egrid2 (cin) from the calculation of cape/cin as grid1 and call the bound to limit range of grid1. I don't see the need of grid1 in the next session code.

Reply:
@SamuelTrahanNOAA Please look at the whole code at
https://github.com/SamuelTrahanNOAA/UPP/blob/ifi/sorc/ncep_post.fd/MISCLN.f#L2102

Reply:
Oh, I see what you're saying. Yes, it looks to me like 2102-2114 are redundant. It must have gotten duplicated in one of my merges.

I will test now to make sure removing that block doesn't change the results.

Reply:
> Oh, I see what you're saying. Yes, it looks to me like 2102-2114 are redundant. It must have gotten duplicated in one of my merges.
> 
> I will test now to make sure removing that block doesn't change the results.

I was going to make the same comments but decided that was ok for now

Reply:
I have removed the duplicate loop and I can confirm the IFI fields don't change.